handbook/upgrade/upgrade.md

```diff
@@ -0,0 +1,312 @@
+
+### Single Node to Multi Node with PostgreSQL
+
+Complete the following steps to upgrade a single node deployment of SystemLink Server to a multi node deployment where the PostgreSQL instance used by the SystemLink Test Monitor service is hosted on a its own server or replica set.
+
+As of SystemLink 21.3.2 SystemLink supports using PostgreSQL as the database backing the Test Monitor service. This configuration is optional, but provides significant performance improvements. `nislmigrate` does not yet support migrating between PostgreSQL servers or replica sets.
+
+The Test Monitor Service performs the migration of test steps, test results, and product from MongoDB to PostgreSQL. If you are using a single node configuration, follow the steps in [**Single Node Migration**](#single-node-migration). If you intend on migrating or upgrading to a new server and utilize a multi node configuration where PostgreSQL is hosted on a dedicated server or replica set, use the following workflow.
+
+1. Backup your SystemLink Server.
+
+1. Attach a volume to store data captured by `nislmigrate`. This will be referenced as `D:\`.
+
+1. Install `nislmigrate`.
+
+1. Run the command `nislmigrate capture --all --secret <your secret> --dir D:\migration`.
+
+1. Detach `D:\`.
+
+    !!!note
+        Since your original SystemLink Server is still operable, any new data created or consumed by SystemLink Server at this time will not be available to your new server.
+
+1. Provision a new Windows server for SystemLink.
+
+1. Provision a PostgreSQL server or replica set.
+
+1. Install and configure SystemLink 21.3.2.
+
+1. Install `nislmigrate` on your new SystemLink Server.
+
+1. Attach the `D:\` volume used to capture data from your original SystemLink server.
+
+1. Run the command `nislmigrate restore --all --secret <your secret> --dir D:\migration`.
+
+1. Open `C:\ProgramData\National Instruments\Skyline\Config\TestMonitor.json`.
+
+1. Change the key `"Postgres.Enabled"` value from `false` to `true`.
+
+1. Add the key `"Postgres.ConnectionString"` and a connection string value. For example:
+
+        "Postgres.ConnectionString" : "Host=<insert db host>;Port=<insert port>;Username=<insert username>;Password=<insert password>;Database=<insert database>;SSL Mode=<Disable or Enable>"
+
+1. Add the key `"Postgres.ConfigurationApproved"=true`.
+
+1. Save and close the file.
+
+1. Open the **NI SystemLink Server Configuration** application.
+
+1. Navigate to **NI SystemLink Service Manager** and click **Restart**.
+
+    !!!note
+        After this step SystemLink will migrate your test steps, results, and products from MongoDB to PostgreSQL. Depending on the size of your data set this process may take some time. The TestMonitor service will display a migration in progress status. If you see an error double check your connection string and restart services if you make any additional changes to the json file.
```

Comment:
progress isn't available in the ServiceManager but is logged to log.txt on the server. Of note I've found that once you're in the section of the migration handling the TestSteps collection any run-rate estimates you perform are pretty accurate. Steps dominate the migration process and the migration is linear in execution (not exponential or logarithmic).

Reply:
I've included these relevant details in the !!!note.